manifests/params.pp

```diff
@@ -1,9 +1,17 @@
 class apt::params {
 
-  if $caller_module_name and $caller_module_name != $module_name {
+  if defined('$caller_module_name') and $caller_module_name and $caller_module_name != $module_name {
     fail('apt::params is a private class and cannot be accessed directly')
   }
 
+  if $::osfamily != 'Debian' {
+    fail('This module only works on Debian or derivatives like Ubuntu')
+  }
+
+  $xfacts = {
+    'lsbdistcodename' => defined('$lsbdistcodename') ? { true => $::lsbdistcodename, default => undef },
```

Comment:
I'd vote for multi-line. It took me way too long to figure out what was happening on the single line...It might get to be very long/a little unwieldy if more facts are added, but at least it should be more readable I think